### PR TITLE
ci: support automation helm chart release

### PR DESCRIPTION
**Pull request by Xunzhuo**
Resolves: #1090

### CODE REVIEW

**Review by haq204 — APPROVED** [review]
lgtm

**Review by AliceProxy — APPROVED** [review]

**Review by zirain — CHANGES_REQUESTED** [review]
/hold

IMO, it's not a good idea adding chart as part of Github release artifacts. 
Hold this before we make a decision about using of OCI.

**Comment by Xunzhuo** [comment]
Right, I am thinking use OCI too. I will update the PRs.

**Comment by zirain** [comment]
```
helm package charts/eg/
helm push eg-0.1.0.tgz oci://docker.io/envoy
```

helm install eg oci://docker.io/envoy/eg --version 0.1.0

**Comment by arkodg** [comment]
sidenote @Xunzhuo  why is the chart called `eg` and not `envoy-gateway` ?

**Comment by Xunzhuo** [comment]
@arkodg This can be changed since we haven`t release helm yet. Any ideas which is better?

File: .github/workflows/latest_release.yaml [hunk 1]
```diff
@@ -62,6 +62,12 @@ jobs:
 
           It is only intended for developers wishing to try out the latest features in Envoy Gateway, some of which may not be fully implemented.
 
+          We use `0.0.0` as the latest chart version to install latest envoy-gateway:
```
Comment (arkodg) [inline .github/workflows/latest_release.yaml:65]: is this common convention used elsewhere ?

File: tools/make/image.mk [hunk 2]
```diff
@@ -118,3 +118,24 @@ push: image.push
 .PHONY: push-multiarch
 push-multiarch: ## Push docker images for multiple platforms to registry.
 push-multiarch: image.multiarch.setup go.build.multiarch image.push.multiarch
+
+
+OCI_REGISTRY ?= oci://docker.io/envoyproxy
```
Comment (arkodg) [inline tools/make/image.mk:123]: can we add these to `helm.mk` ?

File: tools/make/image.mk [hunk 3]
```diff
@@ -118,3 +118,24 @@ push: image.push
...
+CHART_NAME ?= envoy-gateway
+CHART_VERSION ?= 0.1.0
```
Comment (arkodg) [inline tools/make/image.mk:125]: should this be `$(REV)` instead of defaulting to 0.1.0 ?

File: tools/make/image.mk (hunk 2 above)
Comment (arkodg) [reply tools/make/image.mk:123]: or `kube.mk`

**Review by zirain — APPROVED** [review]
LGTM

**Pull request merged**